src/Uno.UI/UI/Xaml/Automation/Peers/FrameworkElementAutomationPeer.Android.cs

```diff
@@ -0,0 +1,105 @@
+using System;
+using Android.Runtime;
+using Android.Text;
+using Android.Views.Accessibility;
+using Uno;
+using Windows.UI.Xaml.Automation.Provider;
+using Windows.UI.Xaml.Controls;
+using Windows.UI.Xaml.Controls.Primitives;
+
+namespace Windows.UI.Xaml.Automation.Peers
+{
+	public partial class FrameworkElementAutomationPeer : AutomationPeer
+	{
+		internal override void SendAccessibilityEvent([GeneratedEnum] EventTypes eventType)
+		{
+			base.SendAccessibilityEvent(eventType);
+
+			switch(eventType)
+			{
+				case EventTypes.ViewClicked:
+					InvokeAutomationPeer();
+					break;
+				default:
+					break;
+			}
+		}
+
+		internal override void OnInitializeAccessibilityNodeInfo(AccessibilityNodeInfo info)
+		{
+			if (!AutomationConfiguration.IsAccessibilityEnabled)
+			{
+				return;
+			}
+
+			var isAccessible = AutomationProperties.GetAccessibilityView(Owner) != AccessibilityView.Raw;
+			if (isAccessible)
+			{
+				info.Focusable = isAccessible;
+				info.ContentDescription = GetName();
+				info.Password = IsPassword();
+				info.Enabled = IsEnabled();
+
+				// TODO: Use GetAutomationControlType() and GetPattern() instead
+				info.Clickable = GetClickable();
+				info.Checked = GetChecked();
+				info.Checkable = GetCheckable();
+				info.Editable = GetEditable();
+				info.InputType = GetInputType();
+				info.MultiLine = GetMultiline();
+				info.Selected = GetSelected();
+				info.MaxTextLength = GetMaxTextLength();
+				info.Scrollable = GetScrollable();
+			}
+		}
+
+		private bool GetScrollable()
+		{
+			return this is IScrollProvider;
+		}
+
+		private int GetMaxTextLength()
+		{
+			return Owner is TextBox textBox ? textBox.MaxLength : 0;
```

Comment:
Why is this not virtual? Should there be a TODO to use ITextProvider?

Reply:
I don't know if there's an equivalent in `AutomationPeer`'s API.

Reply:
I can't find anything about maximum text length at first glance.